src/hb-ot-cmap-table.hh

```diff
@@ -1348,6 +1352,181 @@ struct cmap
 {
   static constexpr hb_tag_t tableTag = HB_OT_TAG_cmap;
 
+  template<typename Iterable1, typename Iterable2,
+           hb_requires (hb_is_iterator (Iterable1)),
+           hb_requires (hb_is_iterator (Iterable2))>
+  bool compare_iterators (Iterable1 Iter_1, Iterable2 Iter_2) const
+  {
+    if (Iter_1.len () != Iter_2.len ())
+      return false;
+
+    bool is_equal = true;
```

Comment:
this variable can be dropped, just return false immediately when you find a not equal pair.